python_modules/dagster/dagster/_core/definitions/assets.py

```diff
@@ -275,33 +273,13 @@ def dagster_internal_init(
         )
 
     def __call__(self, *args: object, **kwargs: object) -> object:
-        from dagster._core.definitions.decorators.op_decorator import DecoratedOpFunction
-        from dagster._core.execution.context.compute import OpExecutionContext
-
         from .graph_definition import GraphDefinition
 
+        # defer to GraphDefinition.__call__ for graph backed assets
         if isinstance(self.node_def, GraphDefinition):
             return self._node_def(*args, **kwargs)
-        op_def = self.op
-        provided_context: Optional[OpExecutionContext] = None
-        if len(args) > 0 and isinstance(args[0], OpExecutionContext):
-            provided_context = _build_invocation_context_with_included_resources(self, args[0])
-            new_args = [provided_context, *args[1:]]
-            return op_def(*new_args, **kwargs)
-        elif (
-            isinstance(op_def.compute_fn, DecoratedOpFunction)
-            and op_def.compute_fn.has_context_arg()
-        ):
-            context_param_name = get_function_params(op_def.compute_fn.decorated_fn)[0].name
-            if context_param_name in kwargs:
-                provided_context = _build_invocation_context_with_included_resources(
-                    self, cast(OpExecutionContext, kwargs[context_param_name])
-                )
-                new_kwargs = dict(kwargs)
-                new_kwargs[context_param_name] = provided_context
-                return op_def(*args, **new_kwargs)
 
-        return op_def(*args, **kwargs)
+        return direct_invocation_result(self, *args, **kwargs)
```

Comment:
nit: Why not just always invoke the `_node_def`? Isn't `OpDefinition` also just calling `direct_invocation_result`?